src/contrib/cluster/Akka.Cluster.Sharding/ShardRegion.cs

```diff
@@ -952,15 +950,16 @@ Address GetNodeAddress(IActorRef shardOrRegionRef)
             try
             {
                 var entityId = getEntityLocation.EntityId;
-                var shardId = _extractShardId(new StartEntity(getEntityLocation.EntityId));
+                var shardId = _messageExtractor.ShardId(getEntityLocation.EntityId,
```

Comment:
Another benefit of this change is that it greatly simplifies shard location queries